lm_eval/models/huggingface.py

```diff
@@ -678,6 +685,36 @@ def tok_decode(self, tokens):
         elif self.AUTO_MODEL_CLASS == transformers.AutoModelForSeq2SeqLM:
             return self.tokenizer.decode(tokens, skip_special_tokens=True)
 
+    def wrap_chat_template(
+        self, requests: List[Instance], generate=False
+    ) -> List[Instance]:
+        """
+        Utility for adding chat templates via the apply_chat_template() method
+        """
+        # TODO: handle repeats > 1 case?
+        # TODO: raise an error if system prompt not compatible with template
+        new_reqs = []
+        for req in requests:
+            context, continuation = req.args[0].strip(), req.args[1]
+            chat = []
+            if self.system_prompt is not None:
+                chat += [{"role": "system", "content": "You are a helpful assistant."}]
```

Comment:
Shouldn't we add the system prompt here?

```suggestion
                chat += [{"role": "system", "content": self.system_prompt}]
```

Reply:
Confirmed, it should be yes. Works fine like that on my fork fyi.

Reply:
I also believe the +='s should be changed to appends, as I noted above.